azure-quantum/azure/quantum/_authentication/_default.py

```diff
@@ -54,162 +52,86 @@ class _DefaultAzureCredential(_ChainedTokenCredential):
-
-        credentials = []  # type: List[TokenCredential]
-        if not self.exclude_token_file_credential:
-            credentials.append(_TokenFileCredential())
-        if not self.exclude_environment_credential:
-            credentials.append(EnvironmentCredential(authority=self.authority))
-        if not self.exclude_managed_identity_credential:
-            credentials.append(ManagedIdentityCredential(client_id=self.managed_identity_client_id))
-        if not self.exclude_shared_token_cache_credential and SharedTokenCacheCredential.supported():
-            try:
-                # username and/or tenant_id are only required when the cache contains tokens for multiple identities
-                shared_cache = SharedTokenCacheCredential(
-                    username=self.shared_cache_username, tenant_id=self.shared_cache_tenant_id, authority=self.authority
-                )
-                credentials.append(shared_cache)
-            except Exception as ex:  # pylint:disable=broad-except
-                _LOGGER.info("Shared token cache is unavailable: '%s'", ex)
-        if not self.exclude_visual_studio_code_credential:
-            credentials.append(VisualStudioCodeCredential(tenant_id=self.vscode_tenant_id))
-        if not self.exclude_cli_credential:
-            credentials.append(AzureCliCredential())
-        if not self.exclude_powershell_credential:
-            credentials.append(AzurePowerShellCredential())
-        if not self.exclude_interactive_browser_credential:
-            credentials.append(InteractiveBrowserCredential(tenant_id=self.interactive_browser_tenant_id))
-        if not self.exclude_device_code_credential:
-            credentials.append(DeviceCodeCredential(tenant_id=self.interactive_browser_tenant_id))
-
+        self._update_authority_and_tenant_id_(
+            arm_base_url=self.arm_base_url,
+            subscription_id=self.subscription_id)
+        credentials = []
+        credentials.append(_TokenFileCredential())
+        credentials.append(EnvironmentCredential())
+        if self.client_id:
+            credentials.append(ManagedIdentityCredential(client_id=self.client_id))
+        if self.authority and self.tenant_id:
+            credentials.append(VisualStudioCodeCredential(authority=self.authority, tenant_id=self.tenant_id))
+            credentials.append(AzureCliCredential(tenant_id=self.tenant_id))
+            credentials.append(AzurePowerShellCredential(tenant_id=self.tenant_id))
+            credentials.append(InteractiveBrowserCredential(authority=self.authority, tenant_id=self.tenant_id))
+            if self.client_id:
+                credentials.append(DeviceCodeCredential(authority=self.authority, client_id=self.client_id, tenant_id=self.tenant_id))
         self.credentials = credentials
 
     def get_token(self, *scopes, **kwargs):
-        # type: (*str, **Any) -> AccessToken
```

Comment:
It looks like we removed the type hinting for the return type. Is this expected?

Reply:
@kikomiss, do you know if the type hints are needed for doc generation?